Example/src/AnimatedScrollExample.js

```diff
@@ -0,0 +1,279 @@
+  { artist: 'The Righteous Brothers', song: "You've Lost That Lovin' Feeling" },
+  { artist: 'The Who', song: 'My Generation' },
+  { artist: 'Martha Reeves and the Vandellas', song: 'Dancing In The Street' },
+  { artist: 'Prince', song: 'When Doves Cry' },
+  { artist: 'Sam Cooke', song: 'A Change Is Gonna Come' },
+  { artist: 'Ike and Tina Turner', song: 'River Deep Mountain High' },
+  { artist: 'The Emotions', song: 'Best Of My Love' },
+];
+
+data = data.slice(0, 10);
+
+const ITEM_SIZE = {
+  size: 250,
+  margin: 70,
+};
+const SCROLL_MARGIN = 20;
+const IPOD_MARGIN = 20;
+const SCREEN_WIDTH =
+  Dimensions.get('window').width - IPOD_MARGIN * 2 - SCROLL_MARGIN * 2;
+const BIG_BALL_SIZE = 200;
+const BIG_BALL_MARGIN = 0;
+const SMALL_BALL_SIZE = 50;
+const INNER_BALL_SIZE =
+  BIG_BALL_SIZE - SMALL_BALL_SIZE * 2 - BIG_BALL_MARGIN * 2;
+const DEFAULT_COVER_URI =
+  'https://e7.pngegg.com/pngimages/950/513/png-clipart-eighth-note-musical-note-stem-notes-music-download-graphic-arts.png';
+
+function ScrollExample() {
+  const position = useSharedValue(0);
+  const animatedRef = useAnimatedRef();
+
+  const itemTotalSize = ITEM_SIZE.size + ITEM_SIZE.margin * 2;
+  const borderMargin = SCREEN_WIDTH / 2 - itemTotalSize / 2 + ITEM_SIZE.margin;
+
+  const scrollToNearestItem = (offset) => {
+    'worklet';
+    let minDistance;
+    let minDistanceIndex = 0;
+    for (let i = 0; i < data.length; ++i) {
+      const distance = Math.abs(i * itemTotalSize - offset);
+      if (minDistance === undefined) {
+        minDistance = distance;
+      } else {
+        if (distance < minDistance) {
+          minDistance = distance;
+          minDistanceIndex = i;
+        }
+      }
+    }
+
+    scrollTo(animatedRef, minDistanceIndex * itemTotalSize, 0, true);
+  };
+
+  const scrollHandler = useAnimatedScrollHandler({
+    onScroll: (e, ctx) => {
+      position.value = e.contentOffset.x;
+    },
+    onEndDrag: (e, ctx) => {
+      scrollToNearestItem(e.contentOffset.x);
+    },
+    onMomentumEnd: (e, ctx) => {
+      scrollToNearestItem(e.contentOffset.x);
+    },
+  });
+
+  const gestureHandler = useAnimatedGestureHandler({
+    onStart: (e, ctx) => {
+      ctx.start = { x: e.x, y: e.y };
+      ctx.last = ctx.start;
+    },
+    onActive: (e, ctx) => {
+      const currentPoz = { x: e.x, y: e.y };
+      const lastPoz = ctx.last;
+      ctx.last = currentPoz;
+      if (currentPoz.x === lastPoz.x && lastPoz.y === currentPoz.y) {
+        // no change so far
+        return;
+      }
+      const changeVector = {
+        x: currentPoz.x - lastPoz.x,
+        y: currentPoz.y - lastPoz.y,
+      };
+      const toCenterV = {
+        x: BIG_BALL_SIZE / 2 - lastPoz.x,
+        y: BIG_BALL_SIZE / 2 - lastPoz.y,
+      };
+      const crossProd =
+        changeVector.x * toCenterV.y - changeVector.y * toCenterV.x;
+      if (crossProd === 0) {
+        return;
+      }
+      const dist = Math.sqrt(changeVector.x ** 2 + changeVector.y ** 2);
```

Comment:
Browsing `Math` module docs I found [`Math.hypot()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/hypot), maybe it'll be useful here?
```js
const dist = Math.hypot(changeVector.x, changeVector.y);
```

Reply:
implemented b6cd607